packages/template-retail-react-app/app/partials/product-view/index.jsx

```diff
@@ -76,161 +76,340 @@ const ButtonWithRegistration = withRegistration(Button)
  * Render a product detail view that includes name, image gallery, price,
  * variant selections, action buttons
  */
-const ProductView = ({
-    product,
-    category,
-    showFullLink = false,
-    imageSize = 'md',
-    isWishlistLoading = false,
-    addToCart,
-    updateCart,
-    addToWishlist,
-    updateWishlist,
-    isProductLoading,
-    isProductPartOfSet = false
-}) => {
-    const intl = useIntl()
-    const history = useHistory()
-    const location = useLocation()
-    const {
-        isOpen: isAddToCartModalOpen,
-        onOpen: onAddToCartModalOpen,
-        onClose: onAddToCartModalClose
-    } = useAddToCartModalContext()
-    const theme = useTheme()
-    const [showOptionsMessage, toggleShowOptionsMessage] = useState(false)
-    const {
-        showLoading,
-        showInventoryMessage,
-        inventoryMessage,
-        quantity,
-        minOrderQuantity,
-        setQuantity,
-        variant,
-        variationParams,
-        variationAttributes,
-        stockLevel,
-        stepQuantity
-    } = useProduct(product, isProductPartOfSet)
-    const canAddToWishlist = !isProductLoading
-    const canOrder =
-        !isProductLoading &&
-        variant?.orderable &&
-        parseInt(quantity) > 0 &&
-        parseInt(quantity) <= stockLevel
+const ProductView = forwardRef(
+    (
+        {
+            product,
+            category,
+            showFullLink = false,
+            imageSize = 'md',
+            isWishlistLoading = false,
+            addToCart,
+            updateCart,
+            addToWishlist,
+            updateWishlist,
+            isProductLoading,
+            isProductPartOfSet = false,
+            variantSelected = () => {},
+            validate = (variant, quantity, stockLevel) =>
+                variant?.orderable && quantity > 0 && quantity <= stockLevel
```

Comment:
Since we're keeping the original logic as the default, do we need to bring in `!isProductLoading` from before?

It used to be:
```
const canOrder =
         !isProductLoading &&
         variant?.orderable &&
         parseInt(quantity) > 0 &&
         parseInt(quantity) <= stockLevel
```

Reply:
Yeah.. I'll add it back for consistency sake. I'm really not a fan of the explicit loading state prop, since you could have been passed a proper product and set it to loading. Personal opinion I suppose.